tests/test_group.rs

```diff
@@ -15,94 +16,146 @@ fn create_commit_optional_path() {
-        alice_update_key_package.clone(),
+        bob_key_package.clone(),
     );
-
-    // Only AddProposals
-    let (commit_mls_plaintext, _welcome_option) = match group_alice_1234.create_commit(
-        group_aad,
-        &alice_credential_bundle,
-        vec![bob_add_proposal.clone()],
-        false,
-    ) {
+    let epoch_proposals = vec![bob_add_proposal];
+    let (mls_plaintext_commit, welcome_bundle_alice_bob_option) = match group_alice_1234
+        .create_commit(
+            group_aad,
+            &alice_credential_bundle,
+            epoch_proposals.clone(),
+            false, /* don't force selfupdate */
+        ) {
         Ok(c) => c,
         Err(e) => panic!("Error creating commit: {:?}", e),
     };
-    let (commit, _confirmation_tag) = match commit_mls_plaintext.content {
-        MLSPlaintextContentType::Commit((commit, confirmation_tag)) => (commit, confirmation_tag),
+    let commit = match &mls_plaintext_commit.content {
+        MLSPlaintextContentType::Commit((commit, _)) => commit,
         _ => panic!(),
     };
     assert!(commit.path.is_none());
 
-    // Only AddProposals with forced self update
-    let (commit_mls_plaintext, _welcome_option) = match group_alice_1234.create_commit(
-        group_aad,
-        &alice_credential_bundle,
-        vec![bob_add_proposal],
-        true,
-    ) {
-        Ok(c) => c,
-        Err(e) => panic!("Error creating commit: {:?}", e),
+    // Alice applies Commit
+    match group_alice_1234.apply_commit(mls_plaintext_commit, epoch_proposals, vec![]) {
+        Ok(_) => {}
+        Err(e) => panic!("Error applying commit: {:?}", e),
     };
+    let ratchet_tree = group_alice_1234.tree().get_public_key_tree();
 
-    let (commit, _confirmation_tag) = match commit_mls_plaintext.content {
-        MLSPlaintextContentType::Commit((commit, confirmation_tag)) => (commit, confirmation_tag),
-        _ => panic!(),
+    // Bob creates group from Welcome
+    let mut group_bob_1234 = match MlsGroup::new_from_welcome(
+        welcome_bundle_alice_bob_option.unwrap(),
+        Some(ratchet_tree),
+        bob_key_package_bundle,
+    ) {
+        Ok(group) => group,
+        Err(e) => panic!("Error creating group from Welcome: {:?}", e),
     };
-    assert!(commit.path.is_some());
 
-    // Own UpdateProposal
+    assert_eq!(
+        group_alice_1234.tree().get_public_key_tree(),
+        group_alice_1234.tree().get_public_key_tree()
+    );
+
+    // Alice updates
+    let alice_update_proposal = group_alice_1234.create_update_proposal(
+        group_aad,
+        &alice_credential_bundle,
+        alice_update_key_package.clone(),
+    );
+    let proposals = vec![alice_update_proposal];
+
+    // Only UpdateProposal
     let (commit_mls_plaintext, _welcome_option) = match group_alice_1234.create_commit(
         group_aad,
         &alice_credential_bundle,
-        vec![alice_update_proposal],
-        true,
+        proposals.clone(),
+        false,
```

Comment:
`/* force self update */`

Reply:
When you use rust-analyzer, you get that for free, that's why I don't make these kind of annotations.